user/tests.py

```diff
@@ -286,3 +307,42 @@ def test_forget_password_no_post(self):
             },
         )
         self.assertEqual(response.status_code, 200)
+
+
+class TestAddPrefView(BaseTest):
+    def test_add_pref_valid(self):
+        self.c.login(username="myuser", password="pass123")
+        Categories.objects.create(category="sushi", parent_category="sushi")
+        Categories.objects.create(category="french", parent_category="french")
+        url = reverse("user:add_preference")
+        form_data = {
+            "pref_list": [
+                "sushi",
+                "french",
+            ]
+        }
+        user_pref_form = UserPreferenceForm(form_data)
+        self.assertTrue(user_pref_form.is_valid())
+        response = self.c.post(path=url, data=form_data)
+        self.assertEqual(response.status_code, 200)
+
+    # def test_add_pref_invalid(self):
```

Comment:
we can delete this since may not be possible to create a bad request since user cannot manually input categories that do not exist in the drop-down menu.